src/main/java/uk/gov/hmcts/reform/em/orchestrator/service/ccdapi/CcdDataApiCaseUpdater.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.hmcts.reform.em.orchestrator.service.ccdapi;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.*;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.em.orchestrator.service.ccdcallbackhandler.CcdCallbackDto;
+import uk.gov.hmcts.reform.em.orchestrator.service.ccdcallbackhandler.CcdCaseDataContent;
+import uk.gov.hmcts.reform.em.orchestrator.service.ccdcallbackhandler.CcdEvent;
+import uk.gov.hmcts.reform.em.orchestrator.service.orchestratorcallbackhandler.CallbackException;
+
+import java.io.IOException;
+
+@Service
+public class CcdDataApiCaseUpdater {
+
+    private final OkHttpClient http;
+    private final AuthTokenGenerator authTokenGenerator;
+    private final String ccdDataBaseUrl;
+    private final String ccdUpdateCasePath = "/cases/%s/events";
+    private final ObjectMapper objectMapper;
+
+    public CcdDataApiCaseUpdater(OkHttpClient http,
+                                 AuthTokenGenerator authTokenGenerator,
+                                 @Value("${ccd.data.api.url}") String ccdDataBaseUrl,
+                                 ObjectMapper objectMapper) {
+        this.http = http;
+        this.authTokenGenerator = authTokenGenerator;
+        this.ccdDataBaseUrl = ccdDataBaseUrl;
+        this.objectMapper = objectMapper;
+    }
+
+    /**
+     * Call to https://hmcts.github.io/reform-api-docs/swagger.html?url=https://hmcts.github.io/reform-api-docs/specs/ccd-data-store-api.v2.json#/case-controller/createEventUsingPOST.
+     *
+     * @param ccdCallbackDto - callbackDTO
+     * @param jwt - authentication
+     */
+    public void executeUpdate(CcdCallbackDto ccdCallbackDto, String jwt) {
+        try {
+            final String json = objectMapper.writeValueAsString(createCcdCaseDataContent(ccdCallbackDto));
+            final RequestBody body = RequestBody.create(json, MediaType.get("application/json"));
+            final Request updateRequest = new Request.Builder()
+                    .addHeader("Authorization", jwt)
+                    .addHeader("experimental", "true")
```

Comment:
what does this do?

Reply:
It allows to use v2 version of the API - Mario said it's safe